numba/cuda/cudaimpl.py

```diff
@@ -405,6 +405,187 @@ def ptx_popc(context, builder, sig, args):
 def ptx_fma(context, builder, sig, args):
     return builder.fma(*args)
 
+
+@lower_cast(types.float16, types.Float)
+def float16_to_float_cast(context, builder, fromty, toty, val):
+    if fromty.bitwidth == toty.bitwidth:
+        return val
+
+    if toty.bitwidth == 32:
+        conv_type = "f32"
+        conv_constraint = "f"
+    elif toty.bitwidth == 64:
+        conv_type = "f64"
+        conv_constraint = "d"
+    else:
+        raise errors.CudaLoweringError(f"Cuda target does not support "
+                                       f"from fp16 converting to float type "
+                                       f"with width {toty.bitwidth}-bits")
+
+    cvt2float_fnty = ir.FunctionType(context.get_value_type(toty),
+                                     [ir.IntType(16)])
+    cvt2float_asm = ir.InlineAsm(cvt2float_fnty,
+                                 f"cvt.{conv_type}.f16 $0, $1;",
+                                 f"={conv_constraint},h")
+    return builder.call(cvt2float_asm, [val])
+
+
+@lower_cast(types.IntegerLiteral, types.float16)
+def literal_int_to_number(context, builder, fromty, toty, val):
+    lit = context.get_constant_generic(
+        builder,
+        fromty.literal_type,
+        fromty.literal_value,
+    )
+    return context.cast(builder, lit, fromty.literal_type, toty)
+
+
```

Comment:
I'm not sure this is necessary - I think this can reuse the `integer_to_float16_cast` below:
```suggestion
```

Reply:
This can be removed when the change in https://github.com/numba/numba/pull/7460#discussion_r763945372 is added. Otherwise, it will confuse the overload-selector.